packages/build/src/plugins_core/pre_dev_cleanup/index.ts

```diff
@@ -18,24 +19,36 @@ const dirExists = async (path: string): Promise<boolean> => {
 const getDirtyDirs = async function ({
   buildDir,
   constants: { INTERNAL_EDGE_FUNCTIONS_SRC, INTERNAL_FUNCTIONS_SRC },
+  packagePath,
 }: CoreStepFunctionArgs): Promise<string[]> {
-  const dirs: string[] = []
+  const directories: string[] = []
 
-  if (INTERNAL_FUNCTIONS_SRC && (await dirExists(resolve(buildDir, INTERNAL_FUNCTIONS_SRC)))) {
-    dirs.push(INTERNAL_FUNCTIONS_SRC)
+  if (INTERNAL_FUNCTIONS_SRC && (await dirExists(resolve(buildDir, packagePath || '', INTERNAL_FUNCTIONS_SRC)))) {
```

Comment:
Have you tested this out in a monorepo? The `INTERNAL_FUNCTIONS_SRC` directory already has the packagePath baked in, see https://github.com/netlify/build/blob/d3ffb0961a7392e46c61486b405d80fbaef2db4d/packages/build/src/core/constants.ts#L127. I believe that we'll have a duplicate `packagePath` segment if we add this in here.

Reply:
There are removal tests also checked in `packages/build/tests/blobs_upload/tests.js` and `packages/build/tests/functions/tests.js` for these folders that include monorepo fixtures that continue to pass. As for testing a monorepo locally, I would need some assistance to set that up.

Reply:
I will add some tests around monorepos and verify these blocks